### DISCUSSION TITLE
Order of children returned by BackendScriptApi.getChildNotes()

### DISCUSSION THREAD

**Discussion by peter-facko** [TriliumNext/Trilium#3801] [General]
Are there any guarantees on the order of children returned by the backend api's `getChildNotes()`?
I assumed it would be the order I see the notes in, but that doesn't seem to be the case.

**Comment by zadam** [comment]
That is strange. `getChildNotes()` should indeed return the notes in the order they are displayed.

**Reply by peter-facko** [reply]
I tried moving the notes in the subtree and now it behaves correctly.
Since I cannot recreate this at the moment, I think we can close this.

**Reply by zadam** [reply]
Unfortunately, it is possible that it gets into a wrongly sorted state and then some other action resorts them. Such bugs are difficult to track down.
